### PR TITLE
add new fieldtypes for dict score modifiers

### PR DESCRIPTION
**Pull request by RaynorChavez**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

* **What is the current behavior?** (You can also link to an open issue here)
No `FieldTypes` defined for map score modifiers

* **What is the new behavior (if this is a feature change)?**
Added 4 new map numeric fieldtypes for dictionary score modifiers. Removed superfluour `__marqo_version__` variable

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: tests/v2_tests/test_score_modifier_search.py [hunk 1]
```diff
@@ -97,7 +97,7 @@ def test_valid_score_modifiers_format(self):
                 "add_to_score": [
                     {"field_name": "add_1", "weight" : -3,
                      },
-                    {"field_name": "add_2", "weight": 1,
+                    {"field_name": "add_2.a", "weight": 1,
                      }]
             }
         self.search_with_score_modifier(score_modifiers=valid_score_modifiers)
```
Comment (papa99do) [inline tests/v2_tests/test_score_modifier_search.py:103]: There's no assertion in this test. what is the expected behaviour?

File: tests/v2_tests/test_score_modifier_search.py [hunk 2]
```diff
@@ -77,11 +77,11 @@ def test_invalid_score_modifiers_format(self):
...
                       "weight": 1,},
-                     {"field_name": "multiply_2",}],
+                     {"field_name": "multiply_2.a",}],
                 "add_to_score": [
                     {"field_name": "add_1", "weight" : 4,
                      },
-                    {"field_name": "add_2", "weight": 1,
+                    {"field_name": "add_2.a", "weight": 1,
                      }]
             }
```
Comment (papa99do) [inline tests/v2_tests/test_score_modifier_search.py:87]: I think the assertion for this test is too wide. we should at least test the error code and some key word in the response if there's any

**Review by papa99do — COMMENTED** [review]
Some minor issue of an existing test. not related to your change.

File: src/marqo/version.py [hunk 3]
```diff
@@ -1,13 +1,8 @@
-__marqo_version__ = "2.9.0"
-__marqo_release_page__ = f"https://github.com/marqo-ai/marqo/releases/tag/{__marqo_version__}"
-
-__minimum_supported_marqo_version__ = "2.6.0"
-
+__minimum_supported_marqo_version__ = "2.9.0"
 
+# NOTE: This isn't used anywhere
```
Comment (papa99do) [inline src/marqo/version.py:3]: So why do we need this if it's not used in any places?

Comment (RaynorChavez) [reply src/marqo/version.py:3]: I didn't totally remove `def supported_marqo_version()` yet as someone might have a comment on its use. 

`__marqo_version__` is removed though.

**Review by papa99do — APPROVED** [review]
LGTM

**Pull request merged**